appscale/tools/remote_helper.py

```diff
@@ -96,22 +96,72 @@ class RemoteHelper(object):
 
 
   @classmethod
-  def start_all_nodes(cls, options, count):
+  def start_all_nodes(cls, options, node_layout):
     """ Starts all nodes in the designated public cloud.
 
     Args:
       options: A Namespace that includes parameters passed in by the user that
         define non-placement-strategy-related deployment options (e.g., keypair
         names, security group names).
-      count: A int, the number of nodes to spawn.
+      node_layout: The node layout of the system including roles.
     Returns:
-      The public IPs and instance IDs (dummy values in non-cloud deployments)
+      The node layout (dummy values in non-cloud deployments)
       corresponding to the nodes that were started.
     """
-    instance_ids, public_ips, private_ips = cls.spawn_nodes_in_cloud(
-      options, count=count)
+    agent = InfrastructureAgentFactory.create_agent(options.infrastructure)
+    params = agent.get_params_from_args(options)
 
-    return instance_ids, public_ips, private_ips
+    # If we have running instances under the current keyname, we try to
+    # re-attach to them. If we have issue finding the locations file or the
+    # IP of the head node, we throw an exception.
+    login_ip = None
+    public_ips, private_ips, instance_ids = agent.describe_instances(params)
+    if public_ips:
+      try:
+        login_ip = LocalState.get_login_host(options.keyname)
+      except (IOError, BadConfigurationException):
+        raise AppScaleException(
+          "Couldn't get login ip for running deployment with keyname"
+          " {}.".format(options.keyname))
+      if login_ip not in public_ips:
+        raise AppScaleException(
+          "Couldn't recognize running instances for deployment with"
+          " keyname {}.".format(options.keyname))
+
+    if login_ip in public_ips:
+      AppScaleLogger.log("Reusing already running instances.")
+      # Set newly obtained node layout info for this deployment.
+      for i, _ in enumerate(instance_ids):
```

Comment:
@whoarethebritons the services might be down but the data is in certain machines only.

Imagine a scenario where the database gets assigned a different IP. Suddenly, there's no data and it's a fresh Cassandra node.

I suggest we put some effort here to handle this case and test it.